### PR TITLE
First deliverable/mini challenge 1 2

### PR DESCRIPTION
**Pull request by DiegooDV**
This is the PR Containing the functionalities:

- Header
- Responsive content
- Loading mock data
- Styling
- Tests
- Coverage reports and (text file coverage report)

### CODE REVIEW

File: src/components/Header/Header.jsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+              <div className="inline-elements">
+                {switchMode ? (
+                  <>
+                    <FontAwesomeIcon className="icon" icon={faMoon} />
+                    <Switch
+                      className="mobile-option"
+                      checked={switchMode}
+                      onChange={handleSwitchMode}
+                    />
+                  </>
```
Comment (jonathanperazawl) [inline src/components/Header/Header.jsx]: You can create a component to wrap this two elements, passing as argument the icon, and onchange method and the checked status: something like:

```jsx
const SwitchItemWrapper = ({ icon, checkedStatus, onChange }) => (
  <>
       <FontAwesomeIcon className="icon" icon={icon} />
       <Switch
            className="mobile-option"
            checked={checkedStatus}
            onChange={onChange}
       />
  </>
)
```

File: src/components/Header/Header.jsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+        <ul className="right-menu inline-elements">
+          {switchMode ? (
+            <>
+              <li className="option" style={{ margin: '0px' }}>
+                <FontAwesomeIcon className="icon" icon={faMoon} />
+              </li>
+              <li className="option" style={{ margin: '0px' }}>
+                <Switch checked={switchMode} onChange={handleSwitchMode} />
+              </li>
```
Comment (jonathanperazawl) [inline src/components/Header/Header.jsx]: Same here, you can create a wrapper component for this section. Please check my previous comment

File: src/components/VideoCard/VideoCard.jsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      </a>
+      <p className="card-date">
+          {moment(props.date).fromNow()}
+        </p>      
+      </footer>
+    </Card>
+  </div>
+);
+
+const splitString = (str, maxChar) => {
```
Comment (jonathanperazawl) [inline src/components/VideoCard/VideoCard.jsx]: You can move this function into an utilities file `src/shared/utils.js`

**Review by jonathanperazawl — COMMENTED** [review]
Hi Diego, I just checked your project and its working fine.

+Thank you for your efforts deploying it to production
I added some suggestions about how can you improve some components, remember to create smaller components that can be re-usable around the project.

On the testing side, if you create smaller components you can avoid duplicating tests that contain these smaller components, and instead of that, you can just validate that the component is rendered without any issues.

File: src/components/Header/Header.jsx (hunk 1 above)
Comment (DiegooDV) [reply src/components/Header/Header.jsx]: Done✅

File: src/components/Header/Header.jsx (hunk 2 above)
Comment (DiegooDV) [reply src/components/Header/Header.jsx]: Done✅

File: src/components/VideoCard/VideoCard.jsx (hunk 3 above)
Comment (DiegooDV) [reply src/components/VideoCard/VideoCard.jsx]: Done✅

**Pull request merged**